playwright/tests/establishment/establishmentWorkflow.spec.ts

```diff
@@ -62,471 +83,52 @@ test.describe("Establishment creation and modification workflow", () => {
     website: faker.internet.domainName(),
     fitForDisabledWorkers: true,
     isEngagedEnterprise: true,
-  } satisfies Partial<FormEstablishmentDto>;
-  test("creates a new establishment", async ({ page }, { retry }) => {
-    await fillEstablishmentFormFirstStep(page, establishments[retry].siret);
-
-    await page
-      .locator(`#${domElementIds.establishment.create.availabilityButton}`)
-      .getByText("Oui")
-      .click();
-
-    await goToNextStep(page, 1, "create");
-    await page
-      .locator(`[for="${domElementIds.establishment.create.searchableBy}-2"]`) // searchable by students
-      .click();
-
-    await goToNextStep(page, 2, "create");
-    await page.fill(
-      `#${domElementIds.establishment.create.businessContact.firstName}`,
-      faker.person.firstName(),
-    );
-    await page.fill(
-      `#${domElementIds.establishment.create.businessContact.lastName}`,
-      faker.person.lastName(),
-    );
-    await page.fill(
-      `#${domElementIds.establishment.create.businessContact.job}`,
-      faker.person.jobType(),
-    );
-    await page.fill(
-      `#${domElementIds.establishment.create.businessContact.phone}`,
-      faker.helpers.fromRegExp(phoneRegexp),
-    );
-    await page.fill(
-      `#${domElementIds.establishment.create.businessContact.email}`,
-      "recette+establishment@immersion-facile.beta.gouv.fr",
-    );
-    await page
-      .locator(
-        `[for='${domElementIds.establishment.create.businessContact.contactMethod}-0']`,
-      )
-      .click();
-
-    await goToNextStep(page, 3, "create");
+  };
+
+  test(
+    "creates a new establishment",
+    createNewEstablishment(
+      initialEstablishmentInformations,
+      establishmentRetries,
+    ),
+  );
 
-    await expect(
-      page.locator(`#${domElementIds.establishment.create.siret}`),
-    ).toHaveValue(establishments[retry].siret);
+  test.describe("Establishment admin", () => {
+    test.use({ storageState: testConfig.adminAuthFile });
 
-    await expect(
-      page.locator(`#${domElementIds.establishment.create.businessName}`),
-    ).not.toHaveValue("");
-    await expect(
-      page.locator(
-        `#${domElementIds.establishment.create.addressAutocomplete}`,
-      ),
-    ).not.toHaveValue("");
-    await expect(
-      page.locator(
-        `#${domElementIds.establishment.create.businessAddresses}-0`,
-      ),
-    ).toHaveValue(establishments[retry].expectedAddress);
-    await page.click(
-      `#${domElementIds.establishment.create.appellations}-add-option-button`,
+    test(
+      "modifies an existing establishment through magic link",
+      modifyEstablishmentMagicLink(updatedInformations, establishmentRetries),
```

Comment:
```suggestion
      updateEstablishmentWithMagicLink(updatedInformations, establishmentRetries),
```
Pour coller avec la version backoffice plus bas

Reply:
J'ai modifié les deux comme suit:

updateEstablishmentThroughMagicLink
updateEstablishmentThroughBackOfficeAdmin